packages/good-design/src/core/inputs/WebVideoUploader.tsx

```diff
@@ -0,0 +1,44 @@
+import { View } from "native-base";
+import React, { useCallback, useRef } from "react";
+import { GoodButton } from "..";
+
+type Props = {
+  onDone: (video?: { base64: string; extension: string }, error?: Error) => Promise<void>;
+};
+
+export const WebVideoUploader = ({ onUpload, isLoading }: { onUpload: Props["onDone"]; isLoading: boolean }) => {
+  const fileRef = useRef<any>(null);
+
+  const handleVideoUpload = useCallback(
+    (event: any) => {
+      event.preventDefault();
+      const imageFile = event.target.files[0];
+      if (imageFile) {
+        console.log({ imageFile });
+        const reader = new FileReader();
+        reader.onload = function (r) {
```

Comment:
we should have utility for readFileAs base64
if not in this package, please get I t form wallet or serve and add here

Reply:
added utility here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/148/commits/b288b2ef6fd887ef9d2c1c6be02e98946bd8737f
implemented in webvideo uploader here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/148/commits/5d185b2ec1e692d5971cbed764ac0751f2094107